bindings/nodejs/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+
+#![allow(clippy::unnecessary_wraps)]
+
+mod classes;
+use classes::*;
+pub mod types;
+
+use neon::prelude::*;
+use once_cell::sync::Lazy;
+use std::{collections::HashMap, sync::Arc};
+use tokio::{runtime::Runtime, sync::RwLock};
+
+pub use iota_wallet::{
+    account::{AccountHandle, SyncedAccount},
+    account_manager::{AccountManager, DEFAULT_STORAGE_FOLDER},
+    actor::{AccountIdentifier, Message as WalletMessage, MessageType, Response, ResponseType, WalletMessageHandler},
+    address::parse as parse_address,
+    event::{
+        on_balance_change, on_broadcast, on_confirmation_state_change, on_error, on_migration_progress,
+        on_new_transaction, on_reattachment, on_stronghold_status_change, on_transfer_progress,
+        remove_balance_change_listener, remove_broadcast_listener, remove_confirmation_state_change_listener,
+        remove_error_listener, remove_migration_progress_listener, remove_new_transaction_listener,
+        remove_reattachment_listener, remove_stronghold_status_change_listener, remove_transfer_progress_listener,
+        EventId,
+    },
+    message::{IndexationPayload, MessageId, RemainderValueStrategy, Transfer, TransferOutput},
+    Error,
+};
+
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
+
+type AccountInstanceMap = Arc<RwLock<HashMap<String, AccountHandle>>>;
+type SyncedAccountHandle = Arc<RwLock<SyncedAccount>>;
+type SyncedAccountInstanceMap = Arc<RwLock<HashMap<String, SyncedAccountHandle>>>;
+
+/// Gets the account instances map.
+fn account_instances() -> &'static AccountInstanceMap {
+    static INSTANCES: Lazy<AccountInstanceMap> = Lazy::new(Default::default);
+    &INSTANCES
+}
+
+pub(crate) async fn get_account(id: &str) -> AccountHandle {
+    account_instances()
+        .read()
+        .await
+        .get(id)
+        .expect("account dropped or not initialised")
+        .clone()
+}
+
+pub(crate) async fn store_account(account_handle: AccountHandle) -> String {
+    let handle = account_handle.clone();
+    let id = handle.id().await;
+
+    account_instances().write().await.insert(id.clone(), account_handle);
+
+    id
+}
+
+/// Gets the synced account instances map.
+fn synced_account_instances() -> &'static SyncedAccountInstanceMap {
+    static INSTANCES: Lazy<SyncedAccountInstanceMap> = Lazy::new(Default::default);
+    &INSTANCES
+}
+
+#[allow(dead_code)]
+pub(crate) async fn get_synced_account(id: &str) -> SyncedAccountHandle {
+    synced_account_instances()
+        .read()
+        .await
+        .get(id)
+        .expect("synced account dropped or not initialised")
+        .clone()
+}
+
+pub(crate) async fn store_synced_account(synced_account: SyncedAccount) -> String {
+    let mut map = synced_account_instances().write().await;
+    let id: String = thread_rng()
+        .sample_iter(&Alphanumeric)
+        .map(char::from)
+        .take(10)
+        .collect();
+    map.insert(id.clone(), Arc::new(RwLock::new(synced_account)));
```

Comment:
Shouldn't we check if the id is already used? I know the probability for this to happen is very low, but not impossible